### PR TITLE
[Serverless] Add dynamodb event for inferred spans

### PR DESCRIPTION
**Pull request by jcstorms1**
### What does this PR do?

- Adds support for DynamoDB inferred spans
- Adds a unit test for SQS that was missed
- Adds a unit test for DynamodDB

### Motivation

Universal Instrumentation

### Describe how to test/QA your changes

Unit Tests 

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/serverless/trace/inferredspan/span_enrichment.go [hunk 1]
```diff
@@ -162,6 +162,33 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithSQSEvent(eventPayload ev
 	}
 }
 
+// EnrichInferredSpanWithDynamoDBEvent uses the parsed event
+// payload to enrich the current inferred span. It applies a
+// specific set of data to the span expected from a DynamoDB event.
+func (inferredSpan *InferredSpan) EnrichInferredSpanWithDynamoDBEvent(eventPayload events.DynamoDBEvent) {
+	eventRecord := eventPayload.Records[0]
+	parsedTableName := strings.Split(eventRecord.EventSourceArn, "/")[1]
```
Comment (kimi-p) [inline pkg/serverless/trace/inferredspan/span_enrichment.go:170]: Nice. Example here `"eventSourceARN": "arn:aws:dynamodb:ap-southeast-2:xxxxxxxxx:table/testProducerTable/stream/2019-07-11T04:32:45.949"`

**Review by kimi-p — APPROVED** [review]
lgtm

File: pkg/serverless/trace/inferredspan/span_enrichment.go [hunk 2]
```diff
@@ -162,6 +162,33 @@ func (inferredSpan *InferredSpan) EnrichInferredSpanWithSQSEvent(eventPayload ev
 	}
 }
 
+// EnrichInferredSpanWithDynamoDBEvent uses the parsed event
+// payload to enrich the current inferred span. It applies a
+// specific set of data to the span expected from a DynamoDB event.
+func (inferredSpan *InferredSpan) EnrichInferredSpanWithDynamoDBEvent(eventPayload events.DynamoDBEvent) {
+	eventRecord := eventPayload.Records[0]
```
Comment (IvanTopolcic) [inline pkg/serverless/trace/inferredspan/span_enrichment.go:169]: Might be a dumb question but are we guaranteed that `Records` is always set, and is always >= 1? Just wondering if there can be some weird AWS case where this isn't true.

File: pkg/serverless/trace/inferredspan/span_enrichment_test.go [hunk 3]
```diff
@@ -201,6 +202,57 @@ func TestEnrichInferredSpanWithSNSEvent(t *testing.T) {
 	assert.True(t, inferredSpan.IsAsync)
 }
 
+func TestEnrichInferredSpanWithSQSEvent(t *testing.T) {
```
Comment (IvanTopolcic) [inline pkg/serverless/trace/inferredspan/span_enrichment_test.go:205]: Is this supposed to be here? Looks like this is a PR for DynamoDB.

**Review by IvanTopolcic — COMMENTED** [review]
Looks good! Two comments.

File: pkg/serverless/trace/inferredspan/span_enrichment_test.go (hunk 3 above)
Comment (jcstorms1) [reply pkg/serverless/trace/inferredspan/span_enrichment_test.go:205]: I saw we had missed it previously so I just added it here

Comment (IvanTopolcic) [reply pkg/serverless/trace/inferredspan/span_enrichment_test.go:205]: Sounds good!

**Review by IvanTopolcic — APPROVED** [review]

File: pkg/serverless/trace/inferredspan/span_enrichment.go (hunk 2 above)
Comment (jcstorms1) [reply pkg/serverless/trace/inferredspan/span_enrichment.go:169]: Yes it is always an array 
```
type DynamoDBEvent struct {
	Records []DynamoDBEventRecord `json:"Records"`
}
```

**Pull request merged**